doc/step-in-groovy.md

```diff
@@ -95,3 +102,19 @@ define a subtype of `GroovyStepExecution` in Java and put those fields there,
 then further subtype that in Groovy and access those fields.
 `GroovyStepExecution` subtypes are trusted, so they can access the fields
 defined in Java code, but untrusted `Jenkinsfile` will not be able to.
+
+## Groovy steps and flow graph
+The way `FlowNode`s are created to record the history of an execution is
+a little different from its Java counterpart.
+
+Invocation of a Groovy step is marked by a block-scoped
+`StepStartNode` and `StepEndNode`, regardless of whether the step
+takes a body closure. All the steps invoked from the Groovy code
+are recorded as flow nodes inside this block.
+
+In contrast, a step written in Java only gets a block-scoped flow
+nodes when it has a body.
+
+When a Groovy step takes a body closure and invokes it, it will not
+automatically create a block-scoped step node. We intend to provide
+a step definition you can use if such block-scoped flow node is desirable.
```

Comment:
My understanding is that this use of the `stage` step is still contested, so I refrained from recommending that to plugin developers.

Since we explain `stage` primarily for users to organize a pipeline into stages, using the same thing for mere grouping of `FlowNode` is confusing for plugin developers. Even if it is the same code internally, I'd think it makes sense to define an alias.

More likely, I'd think we want to leave a different `FlowNode` for this kind of grouping.